src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -156,6 +131,144 @@ class TransportManagementService implements Serializable {
+
+        def responseExtDescriptorUpload = 'responseExtDescriptorUpload.txt'
+
+        def responseCode = script.sh returnStdout: true,
+                                     script: """|#!/bin/sh -e
+                                                | curl ${proxy ? '--proxy ' + proxy + ' ' : ''} \\
+                                                |      --write-out '%{response_code}' \\
+                                                |      -H 'Authorization: Bearer ${token}' \\
+                                                |      -H 'tms-named-user: ${namedUser}' \\
+                                                |      -F 'file=@${file}' \\
+                                                |      -F 'mtaVersion=${mtaVersion}' \\
+                                                |      -F 'description=${description}' \\
+                                                |      --output ${responseExtDescriptorUpload} \\
+                                                |      '${url}/v2/nodes/${nodeId}/mtaExtDescriptors'""".stripMargin()
+
+        return jsonUtils.jsonStringToGroovyObject(getResponseBody(responseExtDescriptorUpload, responseCode, '201', 'MTA Extension Descriptor Upload'))
+    }
+
+    def getNodes(String url, String token) {
+
+        if (config.verbose) {
+            echo("Get nodes started. URL: '${url}'")
+        }
+
+        def parameters = [
+            url          : "${url}/v2/nodes",
+            httpMode     : "GET",
+            contentType  : 'APPLICATION_JSON',
+            customHeaders: [
+                [
+                    maskValue: true,
+                    name     : 'authorization',
+                    value    : "Bearer ${token}"
+                ]
+            ]
+        ]
+
+        def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
+
+        if (proxy){
+            parameters["httpProxy"] = proxy
+        }
+
+        def response = sendApiRequest(parameters)
+        if (response.status != 200) {
+            prepareAndThrowException(response, "Get nodes failed (HTTP status code '${response.status}').")
+        }
+
+        if (config.verbose) {
+            echo("Get nodes successful. Response content '${response.content}'.")
+        }
+
+        return jsonUtils.jsonStringToGroovyObject(response.content)
+    }
+
+    def updateMtaExtDescriptor(String url, String token, Long nodeId, Long idOfMtaExtDescriptor, String file, String mtaVersion, String description, String namedUser) {
+
+        echo("MTA Extension Descriptor Update started.")
+
+        if (config.verbose) {
+        echo("URL: '${url}', NodeId: '${nodeId}', IdOfMtaDescriptor: '${idOfMtaExtDescriptor}', File: '${file}', MtaVersion: '${mtaVersion}'")
+        }
+
+        def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
+
+        def responseExtDescriptorUpdate = 'responseExtDescriptorUpdate.txt'
+
+        def responseCode = script.sh returnStdout: true,
```

Comment:
Is there an explict reason for using `curl` here? That means that we have to run in an environment (--> docker) where `curl` is present. In case there is no special reason for using `curl` I would like to suggest to use the http_request plugin since that is a hard pre-requisite for running piper. I remember cases where we was forced to use `curl` (... but can't provide details, I _think_ it was related to cookies and/or certificates). Why would http_request plugin not work here?

Reply:
> Is there an explict reason for using `curl` here? That means that we have to run in an environment (--> docker) where `curl` is present. In case there is no special reason for using `curl` I would like to suggest to use the http_request plugin since that is a hard pre-requisite for running piper. I remember cases where we was forced to use `curl` (... but can't provide details, I _think_ it was related to cookies and/or certificates). Why would http_request plugin not work here?

Hi Marcus, our APIs(post/put extension descriptor) consume "multipart/form-data". I used http_request plugin initially, but get an error in Jenkins job is as below, 
`for jenkins.plugins.http_request.HttpRequestStep: java.lang.IllegalArgumentException: No enum constant jenkins.plugins.http_request.MimeType.multipart/form-data`
Then I found currently Jenkins http_request plugin does not support a real multipart/form-data as contentType (https://www.jenkins.io/doc/pipeline/steps/http_request/) even though they wrapped uploaded file as multipart entity(https://github.com/jenkinsci/http-request-plugin/blob/eb45bd7e963903f355c22377266fe907c9ea45f7/src/main/java/jenkins/plugins/http_request/HttpRequestExecution.java#L274). However, in our case, there is not only file but also other params, e.g. namedUser, mtaVersion and so on. So it will still not work. 
If my understanding is incorrect, pls just let me know.

Reply:
understood.